cmd/watch.go

```diff
@@ -0,0 +1,214 @@
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			opts.Display = display.Options{
+				Color:                cmdutil.GetGlobalColorization(),
+				ShowConfig:           showConfig,
+				ShowReplacementSteps: showReplacementSteps,
+				ShowSameResources:    showSames,
+				SuppressOutputs:      suppressOutputs,
+				IsInteractive:        interactive,
+				Type:                 display.DisplayWatch,
+				Debug:                debug,
+			}
+
+			return upWorkingDirectory(opts)
+		}),
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&debug, "debug", "d", false,
+		"Print detailed debugging output during resource operations")
+	cmd.PersistentFlags().BoolVar(
+		&expectNop, "expect-no-changes", false,
+		"Return an error if any changes occur during this update")
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"The name of the stack to operate on. Defaults to the current stack")
+	cmd.PersistentFlags().StringVar(
+		&stackConfigFile, "config-file", "",
+		"Use the configuration values in the specified file rather than detecting the file name")
+	cmd.PersistentFlags().StringArrayVarP(
+		&configArray, "config", "c", []string{},
+		"Config to use during the update")
+	cmd.PersistentFlags().BoolVar(
+		&path, "config-path", false,
+		"Config keys contain a path to a property in a map or list to set")
+	cmd.PersistentFlags().StringVar(
+		&secretsProvider, "secrets-provider", "default", "The type of the provider that should be used to encrypt and "+
+			"decrypt secrets (possible choices: default, passphrase, awskms, azurekeyvault, gcpkms, hashivault). Only"+
+			"used when creating a new stack from an existing template")
+
+	cmd.PersistentFlags().StringVarP(
+		&message, "message", "m", "",
+		"Optional message to associate with the update operation")
+
+	// Flags for engine.UpdateOptions.
+	if hasDebugCommands() {
+		cmd.PersistentFlags().StringSliceVar(
+			&policyPackPaths, "policy-pack", []string{},
+			"Run one or more policy packs as part of this update")
+	}
+	cmd.PersistentFlags().BoolVar(
+		&diffDisplay, "diff", false,
+		"Display operation as a rich diff showing the overall change")
+	cmd.PersistentFlags().IntVarP(
+		&parallel, "parallel", "p", defaultParallel,
+		"Allow P resource operations to run in parallel at once (1 for no parallelism). Defaults to unbounded.")
+	cmd.PersistentFlags().BoolVarP(
+		&refresh, "refresh", "r", false,
+		"Refresh the state of the stack's resources before this update")
+	cmd.PersistentFlags().BoolVar(
+		&showConfig, "show-config", false,
+		"Show configuration keys and variables")
+	cmd.PersistentFlags().BoolVar(
+		&showReplacementSteps, "show-replacement-steps", false,
+		"Show detailed resource replacement creates and deletes instead of a single step")
+	cmd.PersistentFlags().BoolVar(
+		&showSames, "show-sames", false,
+		"Show resources that don't need be updated because they haven't changed, alongside those that do")
```

Comment:
For all these various `show*` flags, when is the information shown with `watch`?

Reply:
These can have an effect on which status updates we show during the watch.  For now I've left them, and they will work - but as we finesse the CLI UX we may choose to not support them.